### PR TITLE
KeyTable improvements.

### PR DESCRIPTION
**Pull request by cseed**
Replaced VariantDataset ..._to_pandas with ..._keytable.
Added expand_types and flatten to KeyTable.  These probably need to be more configurable but are a start.
Added KeyTable.toDF.  This allows KeyTables to be easily written to Parquet.

### CODE REVIEW

**Comment by cseed** [comment]
@lfrancioli This allows you to store KeyTables as Parquet.  By default, KeyTable.toDF expands Hail-native types and flattens the schema.

```
vds = hc.import_vcf('/home/cotton/sample.vcf')
(vds.variants_keytable()
  .toDF()
  .write
  .parquet('sample_variants.parquet'))
```

```
$ spark-shell 
scala> spark.sqlContext.read.parquet("sample_variants.parquet").printSchema
root
 |-- v.contig: string (nullable = true)
 |-- v.start: integer (nullable = true)
 |-- v.ref: string (nullable = true)
 |-- v.altAlleles: array (nullable = true)
 |    |-- element: struct (containsNull = true)
 |    |    |-- ref: string (nullable = true)
 |    |    |-- alt: string (nullable = true)
 |-- va.rsid: string (nullable = true)
 |-- va.qual: double (nullable = true)
 |-- va.filters: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- va.pass: boolean (nullable = true)
 |-- va.info.NEGATIVE_TRAIN_SITE: boolean (nullable = true)
 |-- va.info.HWP: double (nullable = true)
 |-- va.info.AC: array (nullable = true)
 |    |-- element: integer (containsNull = true)
 |-- va.info.culprit: string (nullable = true)
 |-- va.info.MQ0: integer (nullable = true)
 |-- va.info.ReadPosRankSum: double (nullable = true)
 |-- va.info.AN: integer (nullable = true)
 |-- va.info.InbreedingCoeff: double (nullable = true)
 |-- va.info.AF: array (nullable = true)
 |    |-- element: double (containsNull = true)
 |-- va.info.GQ_STDDEV: double (nullable = true)
 |-- va.info.FS: double (nullable = true)
 |-- va.info.DP: integer (nullable = true)
 |-- va.info.GQ_MEAN: double (nullable = true)
 |-- va.info.POSITIVE_TRAIN_SITE: boolean (nullable = true)
 |-- va.info.VQSLOD: double (nullable = true)
 |-- va.info.ClippingRankSum: double (nullable = true)
 |-- va.info.BaseQRankSum: double (nullable = true)
 |-- va.info.MLEAF: array (nullable = true)
 |    |-- element: double (containsNull = true)
 |-- va.info.MLEAC: array (nullable = true)
 |    |-- element: integer (containsNull = true)
 |-- va.info.MQ: double (nullable = true)
 |-- va.info.QD: double (nullable = true)
 |-- va.info.END: integer (nullable = true)
 |-- va.info.DB: boolean (nullable = true)
 |-- va.info.HaplotypeScore: double (nullable = true)
 |-- va.info.MQRankSum: double (nullable = true)
 |-- va.info.CCC: integer (nullable = true)
 |-- va.info.NCC: integer (nullable = true)
 |-- va.info.DS: boolean (nullable = true)
```

**Comment by cseed** [comment]
I also added VariantDataset.make_keytable.  With this, one can create keytables with genotype-level values, for example:

```
vds = hc.import_vcf('/home/cotton/sample.vcf')
(vds.make_keytable('v = v, info = va.info', 'gt = g.gt', ['v'])
 .export('test.txt'))
```

test.txt will have a `s.gt` column, one for each sample `s`.

This functionality was already in the Solr and Cassandra export modules, which should now be moved to KeyTable.

The user needs more control over how the KeyTable column names are formed in flatten and make_keytable.

**Comment by jigold** [comment]
Also, I'm not able to see the docs preview. Is this the same problem or new one?

**Comment by danking** [comment]
@cseed was gonna fix that?

**Comment by jigold** [comment]
Also, I see there are no tests for the new features.

**Comment by cseed** [comment]
I fixed the docs preview problem.

**Comment by cseed** [comment]
@jigold @danking Addressed comments.  Can someone take a look?

File: python/pyhail/keytable.py [hunk 1]
```diff
@@ -253,7 +266,7 @@ def key_by(self, key_names):
         :type key_names: list of str
 
         :return: A ``KeyTable`` whose key fields are givne by
```
Comment (jigold) [inline python/pyhail/keytable.py]: small typo

File: python/pyhail/keytable.py [hunk 2]
```diff
@@ -301,3 +359,26 @@ def select(self, field_names):
             return KeyTable(self.hc, self.jkt.select(field_names, new_key_names))
         except Py4JJavaError as e:
             self._raise_py4j_exception(e)
+
+    def toDF(self, expand=True, flatten=True):
```
Comment (jigold) [inline python/pyhail/keytable.py]: Is there a reason this method name is camel case and others are not (example: ``key_by``)

File: python/pyhail/dataset.py [hunk 3]
```diff
@@ -1082,19 +1091,80 @@ def vep(self, config, block_size=None, root=None, force=False, csq=False):
...
+
+        Consider a ``VariantDataset`` ``vds`` with 2 variants and 3 samples::
+
+          Variant	FORMAT	A	B	C
+          1:1:A:T	GT:GQ	0/1:99	./.	0/0:99
+          1:2:G:C	GT:GQ	0/1:89	0/1:99	1/1:93
+
+        Then::
+
+          vds.make_keytable('v = v', 'gt = g.gt', gq = g.gq', [])
```
Comment (jigold) [inline python/pyhail/dataset.py]: I thought we agreed python code would start with `>>>`

**Pull request merged**